### PR TITLE
SHS-5623: Update styles for Date Stacked Vertical Card pattern

### PR DESCRIPTION
**Pull request by mariannuar**
# READY FOR REVIEW

## Summary
Updates for Date Stacked Vertical Card and reuse code

## Need Review By (Date)
05/17

## Urgency
medium

## Steps to Test
1. Visit the following pages on Colorful and Traditional

-  `/mixed-pages/mixed-page-raised-cards`
-  `/collections-cards-continued`

2. Confirm the Date Stacked Vertical Card works and looks as expected on [Figma Option C](https://fourkitchens.clickup.com/t/36718269/SHS-5623#:~:text=UX/Design%20Tickets%20Work)
3. Since we are reusing code for `Vertical Card`, `Vertical Linked Card` and `Date Stacked Vertical Card` please, confirm all of them still look and work as they were before

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)

### CODE REVIEW

File: docroot/themes/humsci/humsci_basic/humsci_basic.theme [hunk 1]
```diff
@@ -266,6 +265,22 @@ function humsci_basic_preprocess_pattern_vertical_card(&$variables) {
...
+ * Implements hook_preprocess_HOOK().
+ */
+function humsci_basic_preprocess_pattern_date_stacked_vertical_card(&$variables) {
+  $renderer = \Drupal::service('renderer');
+  $title_rendered = $renderer->renderPlain($variables['title']);
+
+  if (strpos($title_rendered, '<a') === FALSE) {
+    $original_title = $variables['title'];
+    // If the title is not linked, get the link from the button.
+    _humsci_basic_make_title_the_link($variables);
```
Comment (cienvaras) [inline docroot/themes/humsci/humsci_basic/humsci_basic.theme:278]: Please check line 210 if this file. There's some logic there that's specific to the postcards. You may need to add an extra condition to ensure that there are no issues in other patterns.

File: docroot/themes/humsci/humsci_basic/src/js/shared/linked-cards/linked-cards.js [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
-  const cards = document.querySelectorAll('.hb-vertical-card');
+  const cards = document.querySelectorAll('.hb-vertical-card', '.hb-card--date-stacked', '.hb-vertical-linked-card');
 
   // Loop through each card
   cards.forEach((card) => {
     // Find the main link within each card
-    const mainLink = card.querySelector('.hb-card__title a');
+    let mainLink = '';
+
+    if (card.querySelector('.hb-card__title a')) {
```
Comment (cienvaras) [inline docroot/themes/humsci/humsci_basic/src/js/shared/linked-cards/linked-cards.js:11]: I think you're missing the logic for the date stacked card here.

File: docroot/themes/humsci/humsci_basic/src/js/shared/linked-cards/linked-cards.js [hunk 3]
```diff
@@ -1,11 +1,19 @@
-function verticalCard() {
+function linkedCards() {
```
Comment (cienvaras) [inline docroot/themes/humsci/humsci_basic/src/js/shared/linked-cards/linked-cards.js]: No need to use a named function here. Please switch to using [a closure](https://www.drupal.org/docs/develop/standards/javascript-coding-standards/javascript-coding-standards#strict)

**Review by cienvaras — CHANGES_REQUESTED** [review]
@mariannuar Looks good! There are a couple of small things to review though, please check the comments on the code.

File: docroot/themes/humsci/humsci_basic/humsci_basic.theme (hunk 1 above)
Comment (mariannuar) [reply docroot/themes/humsci/humsci_basic/humsci_basic.theme:278]: @cienvaras I checked and you were right, this piece of code is only happening for postcards. Didn't find issues in other patterns, the condition is working as expected.

**Comment by mariannuar** [comment]
@cienvaras I fixed everything and this is ready for review again. Thanks!

**Review by cienvaras — APPROVED** [review]
Works great, thanks for the fixes! 🚀

**Pull request merged**